src/Resources/contao/library/Contao/Widget.php

```diff
@@ -746,11 +758,41 @@ public function getAttribute($strKey)
 
 
 	/**
+	 * Set a custom value on the widget instead fetching from $_POST values
+	 * when calling validate().
+	 *
+	 * @param mixed $value
+	 *
+	 * @return $this
+	 */
+	public function setCustomValue($value)
+	{
+		$this->varCustomValue = $value;
+		$this->blnCustomValueSet = true;
+
+		return $this;
+	}
+
+	/**
+	 * Will make sure the widget fetches the value from $_POST
+	 *
+	 * @return $this
+	 */
+	public function resetCustomValue()
+	{
+		$this->blnCustomValueSet = false;
+
+		return $this;
+	}
+
+
+	/**
 	 * Validate the user input and set the value
 	 */
 	public function validate()
 	{
-		$varValue = $this->validator($this->getPost($this->strName));
+		$varValue = ($this->blnCustomValueSet) ? $this->varCustomValue : $this->getPost($this->strName);
```

Comment:
The parentheses around `$this->blnCustomValueSet` can be omitted. Also, I'd prefer `blnHasCustomValue`.